### ISSUE TITLE
Automatically check for updated releases

### ISSUE THREAD

**Issue by michaelDCurran** [nvaccess/test_nvdaTracImport#152]
**Reported by mike reiser on 25 Apr 2008 18:04 UTC**
Since NVDA code updates rather rapidly and as I understand it these snapshots are like daily builds of firefox, NVDA should have an automatic update feature like firefox that installs new snapshots as they become available.  The way I see this working out is like this.  

1.  NVDA starts and checks the snapshots page to see if an update is available.  

2.  If an update is available, it asks the user if they would like to download it.  Alternatively it could just go ahead and fetch the update.  

3.  It installs the update and restarts.  I'm not sure if this would be dificult to implament as I'm not a programmer smiles, so someone with more experience could comment.

**Comment by michaelDCurran** [comment]
**Comment by jteh on 20 May 2008 01:00 UTC**
Probably not in 0.6.

**Comment by michaelDCurran** [comment]
**Comment by Bernd on 31 Jul 2009 23:43 UTC**
Should we close this ticket as won'tfix?
I ask because there's now NVDASD to download NVDA Snapshots.

**Comment by michaelDCurran** [comment]
**Comment by jteh on 1 Aug 2009 04:37 UTC**
NVDASD is a great interim solution. However, I'd prefer something more tightly integrated, so let's leave this ticket open. However, it's certainly low priority given that NVDASD exists.

**Comment by michaelDCurran** [comment]
**Modified by jteh on 6 Dec 2010 22:11 UTC**

**Comment by michaelDCurran** [comment]
**Comment by Bernd on 26 Feb 2011 20:36 UTC**
In Notepad there is an updater integrated. Maybe you can integrate it into NVDA. I packed it for you. You can download it here: 
http://dl.dropbox.com/u/11009614/updater.7z

**Comment by michaelDCurran** [comment]
**Comment by orcauser on 30 Nov 2011 23:05 UTC**
Attached diff is a small starting point.

What does it do:
* provide a function that downloads the wiki/Download page and parses for stable release version number.
* Modifies the about nvda message to include this information.

Should it be incorporated into the nvda start screen?

Current flaws:
* If you are running 2011.3 beta/rc, it says your version is up to date, though this can be fixed.
* does not determine newer snapshots.

I am not sure about downloading something on users behalf, at least they should pass via the donate page.
Maybe if a newer release is detected, enable a "go to download" button, that will open the web browser at the correct page, then they can select installer/portable.

Thoughts?
Thanks.

**Comment by michaelDCurran** [comment]
**Comment by orcauser on 30 Nov 2011 23:14 UTC**
:( tabs instead of spaces, will do for next diff.
Sorry

**Comment by michaelDCurran** [comment]
**Comment by jteh on 17 Feb 2012 11:44 UTC**
The initial implementation of this feature will automatically check for and notify of updates, but downloading and installation will be manual. For this reason, it only makes sense to do this for releases, as snapshots get updated almost every day and will therefore just produce useless notifications. This also allows us to gather useful statistics.

Automatic downloading and installation will be considered for a future release.

**Comment by michaelDCurran** [comment]
**Comment by briang1 on 18 Feb 2012 09:27 UTC**
Besides, for snapshots there is a simple to use piece of software to alert you  and download new snapshots if you are so minded.

**Comment by michaelDCurran** [comment]
**Comment by ragb on 18 Feb 2012 12:42 UTC**
I wrote a plugin to report new snapshots, if interessted I can send it. I wrote a simple webservice in python to report that information avoiding flooding nvaccess servers and return pre-made links and such. It can be quicly adapted to stable versions. It is writen in python using the web.py framework and can be put to run anywhere using mod_wsgi. Data is returned in json but other formats can be quickly implemented. Is tehere any interested on this code?

**Comment by michaelDCurran** [comment]
**Comment by jteh on 18 Feb 2012 23:32 UTC**
Replying to [ragb](comment:10):
> I wrote a plugin to report new snapshots
Out of curiosity, what's the point in reporting updated snapshots without automatic installation? They're updated pretty much every day, which makes the notification rather meaningless.

> if interessted I can send it.
Definitely interested in the plugin, particularly if it has a GUI.

> I wrote a simple webservice in python to report that information avoiding flooding nvaccess servers and return pre-made links and such.
The offer is very much appreciated. However, I'm writing one from scratch because we want it to be able to gather statistics, etc. I'm also integrating it with some other areas like the download page to make life easier for us.

**Comment by michaelDCurran** [comment]
**Comment by ragb on 19 Feb 2012 02:26 UTC**
Replying to [jteh](comment:11):
> Replying to [ragb](comment:10):
> > I wrote a plugin to report new snapshots
> Out of curiosity, what's the point in reporting updated snapshots without automatic installation? They're updated pretty much every day, which makes the notification rather meaningless.

Honestly? It is rather pointless to me.. But it seems that some users really appreciate it. The plugin points the user to the direct link of the new snapshot so he don't need to check the snapshots page, find the lin, etc. It was my first plugin, an exercise more than anything else.

> 
> > if interessted I can send it.
> Definitely interested in the plugin, particularly if it has a GUI.

The only GUI component of the pluginis a messagebox propting the user if it wants to download the last snapshot, if on exists. It appears when NVDA is initialized. The plugin is very primitive.
> 
> > I wrote a simple webservice in python to report that information avoiding flooding nvaccess servers and return pre-made links and such.
> The offer is very much appreciated. However, I'm writing one from scratch because we want it to be able to gather statistics, etc. I'm also integrating it with some other areas like the download page to make life easier for us.

YOu can check the web service and some example clients and the nvda plugin at

https://github.com/ragb/nvda_snapshots_ws

There is no documentation, when I wrote that I didn't thought to make it public. In fact I wrote the webservice to have the links to the last snapshots on the nvda portuguese community site, not on plugins or anything else. The nvda_update_ws package contains the cli.py script that, when executed, has commands to track a snapshots branch or update from nvaccess servers. E call cli.py update_snapshots in a cron job to regularly update from nvaccess servers. ws.py contains the real web service and wsgi application. nvdadb.py contains the backend database, a simple sqlite3 db accessed using sqlalchemy .

The nvda plugin is on clients/snapshotVersionChecker.

Regards,

Rui

**Comment by michaelDCurran** [comment]
**Comment by jteh on 28 Feb 2012 04:35 UTC**
Not for 2012.1 after all.

**Comment by michaelDCurran** [comment]
**Comment by jteh on 29 Feb 2012 02:19 UTC**
Bzr branch: http://bzr.nvaccess.org/nvda/updateCheck/

**Comment by michaelDCurran** [comment]
**Modified by jteh on 29 Mar 2012 10:56 UTC**

**Comment by michaelDCurran** [comment]
**Comment by jteh on 5 Apr 2012 04:32 UTC**
Merged in 97bbb1252d3a273aa87b3f8e285154bd01e6877f. Thanks to the work in repackaged, automatic installation is implemented here too. This works for snapshots as well.
